src/app/custom-editor/custom-editor.component.ts

```diff
@@ -172,12 +178,38 @@ export class CustomEditorComponent implements OnInit, OnDestroy, OnChanges {
     this.setValue(value);
     this.editorContentCopy = value;
     this.rawFile = value;
+    this.setContentType();
+    this.setAvailableViews();
     if (value != null || value !== '') {
       this.checkIfTextIsPretty();
     }
     if (this.showPrettifyOnLoad) {
-      this.onViewChange('xml');
+      if (this.contentType == 'xml') {
+        this.onViewChange('xml');
+        return;
+      }
+      if (this.contentType == 'json') {
+        this.onViewChange('json');
+        return;
+      }
+    }
+    this.onViewChange('raw');
+  }
+
+  setContentType(): void {
+    if (this.rawFile.charAt(0) == '<') {
```

Comment:
Unfortunately, a well-formed XML file can also start with a whitespace. So this might not be the most reliable way to check. 
Maybe it would be wise to trim all spaces and then check the first character, but it sounds costly. Maybe check characters until finding a non whitespace character or something like that.

Reply:
Fixed